--- pd-0.37-4/src/g_all_guis.c	2004-09-15 19:03:51.000000000 +0200
+++ pd-0.37-4.patched/src/g_all_guis.c	2005-07-18 14:49:50.147405400 +0200
@@ -353,8 +353,6 @@
     snd = iemgui_raute2dollar(s);
     iemgui->x_snd_unexpanded = snd;
     iemgui->x_snd = snd = canvas_realizedollar(iemgui->x_glist, snd);
-        post("send: before %s, after %s", iemgui->x_snd_unexpanded->s_name,
-    	    iemgui->x_snd->s_name);
 
     iemgui->x_fsf.x_snd_able = sndable;
     iemgui_verify_snd_ne_rcv(iemgui);
