--- u_pdsend.c.org	2005-06-21 10:53:34.000000000 +0200
+++ u_pdsend.c	2005-06-21 10:52:37.000000000 +0200
@@ -105,7 +105,8 @@
         /* now loop reading stdin and sending  it to socket */
     while (1)
     {
-        char buf[BUFSIZE], *bp, nsent, nsend;
+        char buf[BUFSIZE], *bp;
+	unsigned int  nsent, nsend;
         if (!fgets(buf, BUFSIZE, stdin))
             break;
         nsend = strlen(buf);
